wlib/PreferenceDlg.cpp

```diff
@@ -1306,6 +1389,163 @@ class CPrefColorDlg : public CDialogImpl<CPrefColorDlg>,
+						_T("};");
+				}
+				else if (ElementTypeEqual(ATTRIBUTE_TYPE_SALT))
+				{
+					textSample =
+						_T("// SALT sample for illustrating color syntax")	_T("\r\n")
+						_T("/* This is a comment between delimiters */")	_T("\r\n")
+						_T("FIELD:dt_first_seen:RECORDDATE(FIRST)")			_T("\r\n")
+						_T("CONCEPT : locale : +: zip : state : 2, 0")		_T("\r\n")
+						_T("ATTRIBUTEFILE : VEHICLES : ")					_T("\r\n")
+						_T("  NAMED(SALT_Examples.File_Vehicle_Matches_S")	_T("\r\n");
+				}
+				else if (ElementTypeEqual(ATTRIBUTE_TYPE_DUD))
+				{
+					textSample =
+						_T("// DUDE sample for illustrating color syntax")					_T("\r\n")
+						_T("/* This is a comment between delimiters */")					_T("\r\n")
+						_T("NAME RawDataset;")												_T("\r\n")
+						_T("PERMISSIONS")													_T("\r\n")
+						_T("	EDIT:PRIVATE;")												_T("\r\n")
+						_T("	VIEW:PUBLIC;")												_T("\r\n")
+						_T("END")															_T("\r\n")
+						_T("INPUTS")														_T("\r\n")
+						_T("	STRING Name : MAXLENGTH(30);")								_T("\r\n")
+						_T("	RECORD Structure;")											_T("\r\n")
+						_T("	ENUM(CSV,XML,FLAT) Method;")								_T("\r\n")
+						_T("END")															_T("\r\n")
+						_T("OUTPUTS")														_T("\r\n")
+						_T("	DATASET Out1(Structure);")									_T("\r\n")
+						_T("		INT Cnt;")												_T("\r\n")
+						_T("	END")														_T("\r\n")
+						_T("GENERATES INLINE")												_T("\r\n")
+						_T("	%^eOut1% := DATASET(%^qName%,%Structure%,%Method%);")		_T("\r\n")
+						_T("	%C% := COUNT(%Out1%);")										_T("\r\n")
+						_T("ENDGENERATES")													_T("\r\n")
+						_T("VISUALIZE TestPins :TITLE(\"Test pins\")")						_T("\r\n")
+						_T("	CHORO pin2(...TITLE(\"Pins2\"),GEOHASH(pinge...2")			_T("\r\n")
+						_T("END")															_T("\r\n")
+						_T("RESOURCES")														_T("\r\n")
+						_T("	LOGICALFILE File1:FILENAME(\"~thor::temp...20160810\"),")	_T("\r\n")
+						_T("	URL(\"http://10.241.100.159:8010\"),")						_T("\r\n")
+						_T("	ECL Ecl:FILENAME(\"ECL\"),")								_T("\r\n")
+						_T("END")															_T("\r\n");
+				}
+				else if (ElementTypeEqual(ATTRIBUTE_TYPE_ECL))
+				{
+					textSample =
+						_T("// ECL sample for illustrating color syntax")	_T("\r\n")
+						_T("/* This is a comment between delimiters */")	_T("\r\n")
+						_T("import ut;")									_T("\r\n")
+						_T("r := ")											_T("\r\n")
+						_T("  record")										_T("\r\n")
+						_T("   string22 s1 := '123';")						_T("\r\n")
+						_T("   integer4 i1 := 123;")						_T("\r\n")
+						_T("  end;")										_T("\r\n")
+						_T("#option('tmp', true);")							_T("\r\n")
+						_T("d := dataset('tmp::qb', r, thor);")				_T("\r\n")
+						_T("output(d);")									_T("\r\n")
+						_T("Compare:  Added")								_T("\r\n")
+						_T("Compare:  Deleted")								_T("\r\n")
+						_T("Compare:  Changed")								_T("\r\n")
+						_T("Compare:  Moved");
```

Comment:
Your \r\n are all over the place - you could just move them inside the other strings?